sktime/transformations/series/tests/test_temporian.py

```diff
@@ -0,0 +1,170 @@
+        transformer = TemporianTransformer(function=function)
+        X_transformed = transformer.fit_transform(X=X)
+        pd.testing.assert_frame_equal(X_transformed, X + 1)
+
+
+@pytest.mark.skipif(
+    not run_test_for_class(TemporianTransformer),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_hier():
+    """Tests basic function works on Hierarchical."""
+
+    def function(evset):
+        return evset + 1
+
+    examples = get_examples("pd_multiindex_hier", as_scitype="Hierarchical").values()
+    assert len(examples) > 0
+
+    for X in examples:
+        transformer = TemporianTransformer(function=function)
+        X_transformed = transformer.fit_transform(X=X)
+        pd.testing.assert_frame_equal(X_transformed, X + 1)
+
+
+@pytest.mark.skipif(
+    not run_test_for_class(TemporianTransformer),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_compiled():
+    """Tests function already compiled."""
+    import temporian as tp
+
+    @tp.compile
+    def function(evset):
+        return evset + 1
+
+    X = get_examples("pd.Series", as_scitype="Series")[0]
+
+    transformer = TemporianTransformer(function=function, compile=True)
+    X_transformed = transformer.fit_transform(X=X)
+    pd.testing.assert_series_equal(X_transformed, X + 1)
+
+
+@pytest.mark.skipif(
+    not run_test_for_class(TemporianTransformer),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_multiple_output():
+    """Tests function already compiled."""
+
+    def function(evset):
+        split = datetime(1950, 7, 16)
+        return evset.before(split), evset.after(split)
+
+    X = load_airline()
+
+    transformer = TemporianTransformer(function=function)
+    with pytest.raises(TypeError):
+        _ = transformer.fit_transform(X=X)
+
+
+@pytest.mark.skipif(
+    not run_test_for_class(TemporianTransformer),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_multiple_input():
+    """Tests function already compiled."""
+
+    def function(evset, incorrect_param):
+        return evset + 1
+
+    X = load_airline()
+
+    transformer = TemporianTransformer(function=function)
+    with pytest.raises(TypeError):
+        _ = transformer.fit_transform(X=X)
+
+
+@pytest.mark.skipif(
+    not run_test_for_class(TemporianTransformer),
+    reason="run test only if softdeps are present and incrementally (if requested)",
+)
+def test_change_sampling():
+    """Tests function already compiled."""
+    import temporian as tp
+
+    # in this dataset each row is a period of 1 year,
+    # lagging for one year would remove the first row thus changing the sampling
+    X = load_lynx()
+
+    def function(evset):
+        return evset.lag(tp.duration.days(365))
+
+    transformer = TemporianTransformer(function=function, compile=True)
+    with pytest.raises(ValueError):
```

Comment:
What are the messages in these errors? Are they easy to debug? Should we maybe try catching some of them and re-raising with good explanations? At least for the sampling-related ones